### ISSUE TITLE
Incorrect test results

### ISSUE THREAD

**Issue by mysxs** [facebookresearch/av_hubert#45]
The test results are printed out as unk:
![image](https://user-images.githubusercontent.com/88299695/165043638-e4f6236a-bbbd-4568-911d-e45e20f4107a.png)

**Comment by chevalierNoir** [comment]
Can you give more details on testing? For example, what's your decoding command? Which checkpoint did you use?

**Comment by mysxs** [comment]
1、
(1)my finetune command is fairseq-hydra-train --config-dir conf/finetune --config-name base_dy_medium.yaml   task.data=/root/av_hubert/dysarthria/30h_data task.label_dir=/root/av_hubert/dysarthria/30h_data  task.tokenizer_bpe_model=/root/av_hubert/dysarthria/spm650/spm_unigram650.model model.w2v_path=/root/av_hubert/checkpoint/base_lrs3_iter5.pt   hydra.run.dir=/root/av_hubert/dysarthria/medium common.user_dir=`pwd`
medium is my dataset, base_dy_medium.yaml is same as base_lrs3_30h.yaml, excpet modalities are ["video", "audio"].
task.label_dir is my label, spm_unigram650.model is my model 

(2)my decoding command is python -B infer_s2s.py --config-dir ./conf/ --config-name s2s_decode.yaml   dataset.gen_subset=test common_eval.path=/root/av_hubert/dysarthria/mild/checkpoints/checkpoint_best.pt   common_eval.results_path=/root/av_hubert/dysarthria/mild/   override.modalities=['video'] common.user_dir=`pwd`,which medium is my dataset, override.modalities=['audio'] is also.

2、I have the same problem with the lrs3 dataset
my finetune command is fairseq-hydra-train --config-dir conf/finetune/  --config-name base_lrs3_30h.yaml  task.data=/root/av_hubert/data2/30h_data/ task.label_dir=/root/av_hubert/data2/30h_data   task.tokenizer_bpe_model=/root/av_hubert/data2/spm100/spm_unigram100.model model.w2v_path=/root/av_hubert/checkpoint/base_lrs3_iter4.pt   hydra.run.dir=/root/av_hubert/checkpoint common.user_dir=`pwd` optimization.max_epoch=1

my decoding command is python -B infer_s2s.py --config-dir ./conf --config-name s2s_decode.yaml   dataset.gen_subset=test common_eval.path=/root/av_hubert/checkpoint/checkpoints/checkpoint_best.pt common_eval.results_path=/root/av_hubert/data/ common.user_dir=`pwd`  override.modalities=['audio']

**Comment by chevalierNoir** [comment]
Is the decoding result normal if you directly use the fine-tuned checkpoints downloaded [here](https://github.com/facebookresearch/av_hubert#pre-trained-and-fine-tuned-models)? 

Besides, in LRS3 fine-tuning, you were using a config for lip reading, where `task.modalities=["video"]` [here](https://github.com/facebookresearch/av_hubert/blob/main/avhubert/conf/finetune/base_lrs3_30h.yaml#L38). But it is tested under audio-only condition (`override.modalities=['audio']`).

**Comment by mysxs** [comment]
hi， I retest the lrs3 dataset，when I prepare for NO.4 Set up data directory, I ran into a problem：
![image](https://user-images.githubusercontent.com/88299695/165971340-4c062f1c-324a-472d-90e7-20d01f3a1dc3.png)
，my command is python lrs3_manifest.py --lrs3 ${lrs3}, can you help me？

**Comment by chevalierNoir** [comment]
The command is as follows: `python lrs3_manifest.py --lrs3 ${lrs3} --manifest ${lrs3}/file.list  --valid-ids /path/to/valid --vocab-size ${vocab_size}`. Details about the arguments are [here](https://github.com/facebookresearch/av_hubert/tree/main/avhubert/preparation#4-set-up-data-directory).

**Comment by mysxs** [comment]
My command is well written, but still have this problem

**Comment by mysxs** [comment]
I ran the model again with a very small number of lrs3 datasets, there are good changes, try my large dataset tomorrow

![image](https://user-images.githubusercontent.com/88299695/166268963-ba652d84-44a3-4b48-85a8-ccb8797bfecc.png)
![image](https://user-images.githubusercontent.com/88299695/166269041-9d1bc2fb-7bd2-43a8-9af9-38e0d7cae5b5.png)

**Comment by sif99** [comment]
> hi， I retest the lrs3 dataset，when I prepare for NO.4 Set up data directory, I ran into a problem： ![image](https://user-images.githubusercontent.com/88299695/165971340-4c062f1c-324a-472d-90e7-20d01f3a1dc3.png) ，my command is python lrs3_manifest.py --lrs3 ${lrs3}, can you help me？



> hi， I retest the lrs3 dataset，when I prepare for NO.4 Set up data directory, I ran into a problem： ![image](https://user-images.githubusercontent.com/88299695/165971340-4c062f1c-324a-472d-90e7-20d01f3a1dc3.png) ，my command is python lrs3_manifest.py --lrs3 ${lrs3}, can you help me？

@chevalierNoir I have the same error while trying to set up data directory with my own dataset ( for fine-tuning in arabic language)

**Comment by chevalierNoir** [comment]
> > hi， I retest the lrs3 dataset，when I prepare for NO.4 Set up data directory, I ran into a problem： ![image](https://user-images.githubusercontent.com/88299695/165971340-4c062f1c-324a-472d-90e7-20d01f3a1dc3.png) ，my command is python lrs3_manifest.py --lrs3 ${lrs3}, can you help me？
> 
> > hi， I retest the lrs3 dataset，when I prepare for NO.4 Set up data directory, I ran into a problem： ![image](https://user-images.githubusercontent.com/88299695/165971340-4c062f1c-324a-472d-90e7-20d01f3a1dc3.png) ，my command is python lrs3_manifest.py --lrs3 ${lrs3}, can you help me？
> 
> @chevalierNoir I have the same error while trying to set up data directory with my own dataset ( for fine-tuning in arabic language)

This seems to be an issue of training sentencepiece model on your text. Can you print some lines of your label file so that we can try to reproduce it?

**Comment by mysxs** [comment]
> > 你好，我重新测试了lrs3数据集，在准备NO.4的时候设置数据目录，我遇到了一个问题：![图片](https://user-images.githubusercontent.com/88299695/165971340-4c062f1c-324a-472d-90e7-20d01f3a1dc3.png)，我的命令是python lrs3_manifest.py --lrs3 ${lrs3}，你能帮帮我吗？
> 
> > 你好，我重新测试了lrs3数据集，在准备NO.4的时候设置数据目录，我遇到了一个问题：![图片](https://user-images.githubusercontent.com/88299695/165971340-4c062f1c-324a-472d-90e7-20d01f3a1dc3.png)，我的命令是python lrs3_manifest.py --lrs3 ${lrs3}，你能帮帮我吗？
> 
> @chevalierNoir尝试使用自己的数据集设置数据目录时遇到相同的错误（用于阿拉伯语微调）

是代码的问题

**Comment by mysxs** [comment]
> > > 你好，我重新测试了lrs3数据集，在准备NO.4的时候设置数据目录，我遇到了一个问题：![图片](https://user-images.githubusercontent.com/88299695/165971340-4c062f1c-324a-472d-90e7-20d01f3a1dc3.png)，我的命令是python lrs3_manifest.py --lrs3 ${lrs3}，你能帮帮我吗？
> > 
> > 
> > > 你好，我重新测试了lrs3数据集，在准备NO.4的时候设置数据目录，我遇到了一个问题：![图片](https://user-images.githubusercontent.com/88299695/165971340-4c062f1c-324a-472d-90e7-20d01f3a1dc3.png)，我的命令是python lrs3_manifest.py --lrs3 ${lrs3}，你能帮帮我吗？
> > 
> > 
> > @chevalierNoir尝试使用自己的数据集设置数据目录时遇到相同的错误（用于阿拉伯语微调）
> 
> 是代码的问题

lrs3_manifest.py line 39 gen_vocab(Path(f.name), vocab_dir/spm_filename_prefix, 'unigram', args.vocab_size) 不在NamedTemporaryFile里，like this:
![image](https://user-images.githubusercontent.com/88299695/167087278-a9d0fd05-e919-47a7-93aa-6e0be5026b0f.png)

**Comment by YadiraRoCa** [comment]
Hello @sif99, I am also using my own dataset, could you tell me the results you got on train_loss, train_accuracy, val_loss, val_accuracy and the size of the data used? 
Thank you!

**Comment by sif99** [comment]
> Hello @sif99, I am also using my own dataset, could you tell me the results you got on train_loss, train_accuracy, val_loss, val_accuracy and the size of the data used? Thank you!

@YadiraRoCa hello ! Actually I have an error while fine-tuning . 
![image](https://user-images.githubusercontent.com/102472275/167313490-7af9dd03-db15-4032-b2de-bfaf538d8f43.png)
is this the fairseq version that you haves used?

**Comment by YadiraRoCa** [comment]
@sif99 my version is the same (1.0.0a0)

**Comment by mysxs** [comment]
生成的字典里.tsv文件比.wrd文件多个空格，.tsv文件第一行是空格，对应关系否正确?
![image](https://user-images.githubusercontent.com/88299695/167623372-da04fabd-98a9-4f60-8dcd-2c4addff2365.png)
![image](https://user-images.githubusercontent.com/88299695/167623428-f9bc3f87-ba23-4941-aa32-c98cd2f8343a.png)
![image](https://user-images.githubusercontent.com/88299695/167623455-dbbc2de4-f939-47c1-8a63-7483c7bbf30f.png)

**Comment by chevalierNoir** [comment]
> 生成的字典里.tsv文件比.wrd文件多个空格，.tsv文件第一行是空格，对应关系否正确? ![image](https://user-images.githubusercontent.com/88299695/167623372-da04fabd-98a9-4f60-8dcd-2c4addff2365.png) ![image](https://user-images.githubusercontent.com/88299695/167623428-f9bc3f87-ba23-4941-aa32-c98cd2f8343a.png) ![image](https://user-images.githubusercontent.com/88299695/167623455-dbbc2de4-f939-47c1-8a63-7483c7bbf30f.png)

Right. First line of tsv file is `/`.

**Comment by mysxs** [comment]
> Hello @sif99, I am also using my own dataset, could you tell me the results you got on train_loss, train_accuracy, val_loss, val_accuracy and the size of the data used? Thank you!

我用了2000多的数据集，比如epoch=181时，各项参数如下： epoch": 181, "train_loss": "5.316", "train_nll_loss": "0.238", "train_total": "30.8446", "train_n_correct": "30.6351", "train_ppl": "1.18", "train_accuracy": "99.321", "train_wps": "76.7", "train_ups": "2.49", "train_wpb": "30.8", "train_bsz": "8.5", "train_num_updates": "26836", "train_lr": "8.03143e-05", "train_gnorm": "2.17", "train_loss_scale": "256", "train_train_wall": "18", "train_gb_free": "7.3", "train_wall": "12587

**Comment by mysxs** [comment]
> Hello @sif99, I am also using my own dataset, could you tell me the results you got on train_loss, train_accuracy, val_loss, val_accuracy and the size of the data used? Thank you!

"epoch": 182, "valid_loss": "20.731", "valid_nll_loss": "5.402", "valid_total": "27.7115", "valid_n_correct": "10.0769", "valid_ppl": "42.28", "valid_accuracy": "36.364", "valid_wps": "75", "valid_wpb": "27.7", "valid_bsz": "7.9", "valid_num_updates": "26984", "valid_best_accuracy": "37.058

**Comment by YadiraRoCa** [comment]
Hi @mysxs ! Thank you for your answer.  I'm doing fine-tuning with a dataset of over 4000 sentences which isn't in the English language. Is your dataset in English?  Could you tell me if this result was obtained by base_lrs3_iter4.pt ? What configuration file of finetune section do you use?

**Comment by mysxs** [comment]
> Hi @mysxs ! Thank you for your answer. I'm doing fine-tuning with a dataset of over 4000 sentences which isn't in the English language. Is your dataset in English? Could you tell me if this result was obtained by base_lrs3_iter4.pt ? What configuration file of finetune section do you use?

hi～ yes，my dataset in English，but my dataset is pathology speech。一部分数据集的wer大约是52%，一部分数据集的wer大约是90%，现在我还在做单个说话人的相关实验。
我用到的finetune file is base_lrs3_30h.yaml

**Comment by JinChow** [comment]
> > Hi @mysxs ! Thank you for your answer. I'm doing fine-tuning with a dataset of over 4000 sentences which isn't in the English language. Is your dataset in English? Could you tell me if this result was obtained by base_lrs3_iter4.pt ? What configuration file of finetune section do you use?
> 
> hi～ yes，my dataset in English，but my dataset is pathology speech。一部分数据集的wer大约是52%，一部分数据集的wer大约是90%，现在我还在做单个说话人的相关实验。 我用到的finetune file is base_lrs3_30h.yaml

       @mysxs ， 您好，我现在也在复现这篇ReVISE的工作，我想在我自己的数据集上进行微调，但是我遇到了一些问题，能请教一些您嘛，已经迷茫好几天了，恳请您指点，2750351250@qq.com，这是我的邮箱，希望能获得您的联系方式，科研小白感激不尽！

**Comment by RayShing** [comment]
> > > Hi @mysxs ! Thank you for your answer. I'm doing fine-tuning with a dataset of over 4000 sentences which isn't in the English language. Is your dataset in English? Could you tell me if this result was obtained by base_lrs3_iter4.pt ? What configuration file of finetune section do you use?
> > 
> > 
> > hi～ yes，my dataset in English，but my dataset is pathology speech。一部分数据集的wer大约是52%，一部分数据集的wer大约是90%，现在我还在做单个说话人的相关实验。 我用到的finetune file is base_lrs3_30h.yaml
> 
> ```
>    @mysxs ， 您好，我现在也在复现这篇ReVISE的工作，我想在我自己的数据集上进行微调，但是我遇到了一些问题，能请教一些您嘛，已经迷茫好几天了，恳请您指点，2750351250@qq.com，这是我的邮箱，希望能获得您的联系方式，科研小白感激不尽！
> ```

Hello, I am also interested in reproducing ReVISE. Did u successfully finish it? Thanks!
